packages/slice-machine/src/components/PageLayout/PageLayout.tsx

```diff
@@ -16,6 +16,7 @@ export const PageLayout: FC<PageLayoutProps> = ({
     {children}
     <div
       className={clsx(styles.borderTop, styles.borderTopColor[borderTopColor])}
+      data-cy="top-border"
```

Comment:
⚠️ Maybe a bit more details?

Something like `app-top-border` or `window-top-border`?

Reply:
> UI components shouldn't contain domain/business/app concerns in my opinion.

I agree as it makes components reusable.

> I think we should assign data-cy/data-testid through spread props in the AppLayout component. That way, we could very well move PageLayout to the Editor's design system in the future.

Is `data-cy`/`data-testid` something we need to make available to all components? If so, should we be adding the prop to all of our `Prop` types?

We could make a `TestingProps` type that we could intersect with our other types:

```ts
type ExampleProps = FC<PropsWithChildren & TestingProps>
```

In this specific case, we can't spread props to the top border component since it isn't the top-level element. We could implement something like a `borderTopProps` prop, but that feels inelegant. Perhaps we need to move the border out to its own component?

Reply:
I think anyway we need that management with TestingProps I needed it also for the Add IconButton within `WindowTabsList`. We can also find other ways but I guess we need a way to pass testing props to pure generic UI component. WDYT @bapmrl?

Reply:
@bapmrl With `...otherProps`, wouldn't we still need to type the props with `data-cy: string`? It would also need to be added up the tree to top-level consumers.

This feels dirty, but we could globally add `data-cy` as a prop to `JSX.ElementType`:

```ts
declare namespace JSX JSX {
  interface ElementClass {
    "data-cy"?: string;
  }
}
```

Then when we use `...othersProps`, it should pass down automatically.

Reply:
I did that @angeloashmore btw for the moment:

<img width="619" alt="Screenshot 2023-12-19 at 20 27 40" src="https://github.com/prismicio/slice-machine/assets/19946868/e6c96671-eabb-4418-b481-8fd091d3144f">

WDYT about something like this, the name is not known then?

Reply:
So here, I want to be able to target directly the icon button from within `WindowTabsList`. 

For this specific use case, the best will be to add a label to the button, as Angelo suggested in the past. My point is to show that if you want to target a specific nested component with a test id, you will need to find a way to pass the id without mentioning in props the "data-cy" for example.

Reply:
I solved this specific issue by changing the extra `<div>` used to draw the top border into an actual border on the layout's `<div>`. As a result, there is no need to locate a child element.

However, this is still a general issue that would need to be addressed eventually.

I prefer generalizing how we handle test props using something like `...otherProps` (so we don't hard-code what a test prop is) or `testingProps?: Record<string, unknown>` (so that we can isolate test-specific props and potentially remove them in prod).

Using a convention like `testingProps` allows us to extend it to child components, like `iconButtonTestingProps`.

```tsx
<WindowTabsList
  onAddNewTab={() => console.log("Added new tab")}
  testingProps={{ "data-cy": "foo" }}
  iconButtonTestingProps={{ "data-cy": "bar" }}
>
  {/* ... */}
</WindowTabsList>
```